packages/client/src/connection.js

```diff
@@ -41,12 +40,44 @@ class Connection {
  * @template {Record<string, any>} T
  * @template {API.Tuple<API.ServiceInvocation<C, T>>} I
  * @param {API.Connection<T>} connection
- * @param {I} invocations
- * @returns {Promise<API.InferServiceInvocations<I, T>>}
+ * @param {I} workflow
+ * @returns {Promise<API.InferWorkflowReceipts<I, T>>}
  */
-export const execute = async (invocations, connection) => {
-  const request = await connection.encoder.encode(invocations, connection)
+export const execute = async (workflow, connection) => {
+  const request = await connection.codec.encode(workflow, connection)
   const response = await connection.channel.request(request)
-  const result = await connection.decoder.decode(response)
-  return result
+  // We may fail to decode the response if content type is not supported
+  // or if data was corrupted. We do not want to throw in such case however,
+  // because client will get an Error object as opposed to a receipt, to retain
+  // consistent client API with two kinds of errors we encode caught error as
+  // a receipts per workflow invocation.
+  try {
+    return await connection.codec.decode(response)
+  } catch (error) {
+    const { message, ...cause } = /** @type {Error} */ (error)
```

Comment:
I believe technically `error` could be undefined and the destructuring would lead to a `SyntaxError` in that case. (might be wise to check `typeof error === 'object'` first.
unlikely but a pain if/when it does.

Reply:
```suggestion
    // No third party code is run during decode and we know
    // we only throw an Error
    const { message, ...cause } = /** @type {Error} */ (error)
```